lib/ret_web/views/api/v1/avatar_view.ex

```diff
@@ -10,23 +10,41 @@ defmodule RetWeb.Api.V1.AvatarView do
     %{avatars: [render_avatar(avatar)]}
   end
 
-  def render_avatar(avatar) do
+  def render_avatar(%Avatar{} = a) do
     %{
-      avatar_id: avatar.avatar_sid,
-      parent_avatar_id: unless(is_nil(avatar.parent_avatar), do: avatar.parent_avatar.avatar_sid),
-      parent_avatar_listing_id:
-        unless(is_nil(avatar.parent_avatar_listing), do: avatar.parent_avatar_listing.avatar_listing_sid),
-      name: avatar.name,
-      description: avatar.description,
-      attributions: if(is_nil(avatar.attributions), do: %{}, else: avatar.attributions),
-      allow_remixing: avatar.allow_remixing,
-      allow_promotion: avatar.allow_promotion,
-      gltf_url: avatar |> Avatar.gltf_url(),
-      base_gltf_url: avatar |> Avatar.base_gltf_url(),
+      avatar_id: a.avatar_sid,
+      parent_avatar_id: unless(is_nil(a.parent_avatar), do: a.parent_avatar.avatar_sid),
+      parent_avatar_listing_id: unless(is_nil(a.parent_avatar_listing), do: a.parent_avatar_listing.avatar_listing_sid),
+      name: a.name,
+      description: a.description,
+      attributions: if(is_nil(a.attributions), do: %{}, else: a.attributions),
+      allow_remixing: a.allow_remixing,
+      allow_promotion: a.allow_promotion,
+      gltf_url: a |> Avatar.gltf_url(),
+      base_gltf_url: a |> Avatar.base_gltf_url(),
       files:
         for col <- Avatar.file_columns(), into: %{} do
           key = col |> Atom.to_string() |> String.replace_suffix("_owned_file", "")
-          {key, avatar |> Avatar.file_url_or_nil(col)}
+          {key, a |> Avatar.file_url_or_nil(col)}
+        end
+    }
+  end
+
+  def render_avatar(%AvatarListing{} = a) do
+    %{
+      avatar_id: a.avatar_listing_sid,
+      parent_avatar_listing_id: unless(is_nil(a.parent_avatar_listing), do: a.parent_avatar_listing.avatar_listing_sid),
+      name: a.name,
+      description: a.description,
+      attributions: if(is_nil(a.attributions), do: %{}, else: a.attributions),
+      allow_remixing: a.avatar.allow_remixing,
+      allow_promotion: a.avatar.allow_promotion,
+      gltf_url: a |> AvatarListing.gltf_url(),
+      base_gltf_url: a |> AvatarListing.base_gltf_url(),
+      files:
+        for col <- Avatar.file_columns(), into: %{} do
+          key = col |> Atom.to_string() |> String.replace_suffix("_owned_file", "")
+          {key, a |> AvatarListing.file_url_or_nil(col)}
```

Comment:
Would be good to follow the form we have in `scene_view`, where we refactor common fields where possible. Comparing the two, it also seems you're missing guards for preventing views for removed or delisted avtars.

Reply:
Hmm agree this should be accounted for, but seems odd to do it in the view instead of the controller. Will add checks to the controller.

Reply:
I think the reason we did it in the view for scenes is because `render_scene` is used from `hub_view`. Doesn't apply to avatars for now, I guess.